src/Tempest/Console/src/Components/Interactive/TaskComponent.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Components\Interactive;
+
+use Closure;
+use Generator;
+use RuntimeException;
+use Symfony\Component\Process\Process;
+use Tempest\Console\Components\ComponentState;
+use Tempest\Console\Components\Concerns\HasErrors;
+use Tempest\Console\Components\Concerns\HasState;
+use Tempest\Console\Components\Renderers\SpinnerRenderer;
+use Tempest\Console\Components\Renderers\TaskRenderer;
+use Tempest\Console\InteractiveConsoleComponent;
+use Tempest\Console\Terminal\Terminal;
+use Throwable;
+use function Tempest\Support\arr;
+
+final class TaskComponent implements InteractiveConsoleComponent
+{
+    use HasErrors;
+    use HasState;
+
+    private TaskRenderer $renderer;
+
+    private int $processId;
+
+    private float $startedAt;
+
+    private ?float $finishedAt = null;
+
+    private array $sockets;
+
+    private array $log = [];
+
+    private(set) array $extensions = ['pcntl'];
+
+    public function __construct(
+        readonly string $label,
+        private null|Process|Closure $handler = null,
+    ) {
+        $this->handler = $this->resolveHandler($handler);
+        $this->renderer = new TaskRenderer(new SpinnerRenderer(), $label);
+        $this->startedAt = hrtime(as_number: true);
+    }
+
+    public function render(Terminal $terminal): Generator
+    {
+        // If there is no task handler, we don't need to fork the process, as
+        // it is a time-consuming operation. We can simply consider it done.
+        if ($this->handler === null) {
+            $this->state = ComponentState::SUBMITTED;
+
+            yield $this->renderTask($terminal);
+
+            return true;
+        }
+
+        $this->sockets = stream_socket_pair(domain: STREAM_PF_UNIX, type: STREAM_SOCK_STREAM, protocol: STREAM_IPPROTO_IP);
+        $this->processId = pcntl_fork();
```

Comment:
Couldn't we use `symfony/process` instead? It's already a dependency because we use for async commands, and it has cross-platform support.

Reply:
The process component can't execute closures, can it?

Reply:
Aah, that's true! Could fibers be an alternative? Or do you want to run tasks in parallel?

Reply:
I tried fibers because I really didn't want to use `pcntl`, but it doesn't work because yeah, the task needs to run in parallel for the spinner to spin 😔